gui/app/services/effectHelperService.js

```diff
@@ -163,16 +143,62 @@
 
             // This checks if an item is in the effect.group array and returns true.
             // This allows us to check boxes when loading up this button effect.
-            $scope.groupCheckboxer = function (group){
-              if($scope.effect.groups != null) {
-                return $scope.effect.groups.indexOf(group) != -1;
-              } else {
-                return false;
-              }              
+            $scope.groupCheckboxer = function (list, item){
+              return service.checkSavedArray(list, item);         
+            }
+
+          };
+          break;
+
+        case EffectType.COOLDOWN:
+          controller = ($scope, boardService) => {
+          
+            // Get all control id's in an array so we can add checkboxes.
+            $scope.boardButtons = boardService.getControlIdsForSelectedBoard();
+
+            // This sets the effect.buttons to an array of checked items.
+            $scope.buttonArray = function(list, item){
+              $scope.effect.buttons = service.getCheckedBoxes(list, item);
+            }
+
+            // This checks if an item is in the effect.buttons array and returns true.
+            // This allows us to check boxes when loading up this button effect.
+            $scope.buttonCheckboxer = function (list, item){
+              return service.checkSavedArray(list, item);         
+            }
+            
+          };
+          break;
+
+        case EffectType.CELEBRATION:
+          controller = ($scope) => {
+
+            $scope.celebrationTypes = [
+              "Fireworks"
+            ]
+
+          };
+          break;
+
+        case EffectType.CUSTOM_SCRIPT:
+          controller = ($scope) => {
+
+            // Grab files in folder when button effect shown.
+            $scope.scriptArray = fs.readdirSync('./user-settings/scripts');
+
+            // Grab files in folder on refresh click.
+            $scope.getNewScripts = function (){
+              $scope.scriptArray = fs.readdirSync('./user-settings/scripts');
+            }
+
+            // Open script folder on click.
+            $scope.openScriptsFolder = function(){
+              ipcRenderer.send('openScriptsFolder');
```

Comment:
At the moment, this is entirely sufficient but we eventually might want to move event sending to the listenerService (and rename that to the eventService) so we have once consolidated place for sending and listening to events.